src/tdastro/astro_utils/opsim.py

```diff
@@ -301,6 +301,49 @@ def write_opsim_table(self, filename, tablename="observations", overwrite=False)
 
         con.close()
 
+    def time_bounds(self):
+        """Returns the min and max times for all observations in the OpSim.
+
+        Returns
+        -------
+        t_min, t_max : float, float
+            The min and max times for all observations in the OpSim.
+        """
+        t_min = self.table[self.colmap["time"]].min()
+        t_max = self.table[self.colmap["time"]].max()
+        return t_min, t_max
+
+    def filter_rows(self, rows):
```

Comment:
I'd prefer to return a new OpSim object here.

Reply:
Done.